projects/project2/com.test.firstscalaprogram/src/com/test/firstscalaprogram/MainObject.scala

```diff
@@ -0,0 +1,206 @@
+package com.test.disdupProject
+import org.apache.spark._
+import org.apache.spark.rdd.RDD
+import scala.collection.mutable.HashMap
+import scala.collection.mutable.ListBuffer
+import scala.xml._
+import shapeless.ops.tuple.Length
+
+
+object MainObject {
+  
+  val pathOfDataFile : String = "E:\\tmp\\Spark_Workspace\\CommaSeperatedData.txt"
+  val appName : String = "Disdup"
+  val outputDirectory : String = "E:\\tmp\\Spark_Workspace\\output"
+  
+  
+ def main(args:Array[String]) =
+ { 
+   try
+   {  
+     NaiveDistinct()
+     //BlockBased()
+     //partitionEqually()
+   }
+   catch
+   {
+     case 
+     e : Exception => println(e)
+     e.printStackTrace()
+   }
+ }
+ 
+ 
+ def NaiveDistinct()
+ {
+   try
+   {
+    val conf:SparkConf = new SparkConf().setAppName("Disdup").setMaster("local[*]")
+    val sc = SparkContext.getOrCreate(conf)
+    
+    val startTime = System.currentTimeMillis();
+    val textFile = sc.textFile("E:\\tmp\\Spark_Workspace\\CommaSeperatedData.txt", 2)
+    
+    ////////////////////Naive Distinct Function////////////////
+    
+    val testing = textFile.flatMap(record => 
+      {
+        record.split("\n")  
+      }
+      ).distinct()
+    testing.saveAsTextFile("E:\\tmp\\Spark_Workspace\\output")
+    
+    ////////////////////////////////////////////////////////
+    
+    val endTime = System.currentTimeMillis();
+    val processingTime = endTime - startTime
+    println("Time taken: "+processingTime)
+   
+   }
+   catch
+   {
+     case 
+     e : Exception => println(e)
+     e.printStackTrace()
+   }
+ }
+ 
+ 
+ def partitionEqually()
+ {
+   try
+   {
+    val conf:SparkConf = new SparkConf().setAppName(appName).setMaster("local[*]")
```

Comment:
Why not using `SparkSession` instead?

Reply:
val spark = SparkSession.builder.appName("appName").master("local[*]").getOrCreate()

Replaced, will update it.